File: src/test/java/koorkah/TestRange.java
package koorkah;

import org.junit.jupiter.api.Test;

class TestRange extends BaseT
{

    @Test
    void testSimple()
    {
        String expression = koorkah.toRegEx("range{`a`:`c`}");
        assertRegexIs("[a-c]", expression);
        assertRegexMatchesText(expression, "a");
        assertRegexMatchesText(expression, "c");
        assertRegexDoesntMatchText(expression, "f");
        assertRegexDoesntMatchText(expression, "2");      
    }
    
    @Test
    void testComplex()
    {
        String expression = koorkah.toRegEx("range{`a`:`c`, `6`:`9`}");
        assertRegexIs("[a-c6-9]", expression);
        assertRegexMatchesText(expression, "a");
        assertRegexMatchesText(expression, "7");
        assertRegexDoesntMatchText(expression, "f");
        assertRegexDoesntMatchText(expression, "2");      
    }

}

File: src/python/test/TestRange.py
import unittest
from Utils import Utils 
from Refi import Refi


class TestRange(unittest.TestCase):

    def setUp(self):
        self.refi = Refi()
        self.utils = Utils()


    def test_Simple(self):

        expression = self.refi.toRegEx("range{`a`:`c`}")
        self.utils.assertRegexIs("[a-c]", expression)
        self.utils.assertRegexMatchesText(expression, "a")
        self.utils.assertRegexMatchesText(expression, "c")
        self.utils.assertRegexDoesntMatchText(expression, "f")
        self.utils.assertRegexDoesntMatchText(expression, "2")      
    
    def test_Complex(self):

        expression = self.refi.toRegEx("range{`a`:`c`, `6`:`9`}")
        self.utils.assertRegexIs("[a-c6-9]", expression)
        self.utils.assertRegexMatchesText(expression, "a")
        self.utils.assertRegexMatchesText(expression, "7")
        self.utils.assertRegexDoesntMatchText(expression, "f")
        self.utils.assertRegexDoesntMatchText(expression, "2")      

    def runTest(self):
        pass  

if __name__ == '__main__':
    unittest.main()    
    


